### PR TITLE
Clean up procuration controllers

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

File: app/config/services/community.xml [hunk 1]
```diff
@@ -77,9 +77,7 @@
         </service>
 
         <!-- Procuration -->
-        <service id="app.procuration.request_flow" class="AppBundle\Procuration\ProcurationRequestFlow">
-            <argument type="service" id="session" />
-        </service>
+        <service autowire="true" id="AppBundle\Procuration\ProcurationRequestSession" />
```
Comment (B-Galati) [inline app/config/services/community.xml]: You can add this declaration to `services.xml` if you want. 
I created this file to be our autowire/autoconfig that is shipped with SF by default now.
The intention is to migrate namespace by namespace as we developed.

File: src/Controller/EnMarche/ProcurationController.php [hunk 2]
```diff
@@ -37,11 +37,13 @@ public function landingAction(Request $request): Response
      */
     public function indexAction(Request $request): Response
     {
-        $this->getProcurationFlow()->reset();
+        $procurationSession = $this->get(ProcurationRequestSession::class);
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: You can use Action injection instead

File: src/Controller/EnMarche/ProcurationController.php [hunk 3]
```diff
@@ -37,11 +37,13 @@ public function landingAction(Request $request): Response
...
+        // clear previous session if any
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: Why ?
By the way the comment does not bring more information that the method name.

File: templates/procuration_manager/request.html.twig [hunk 4]
```diff
@@ -17,12 +17,12 @@
                 Désassocier de {{ request.foundProxy.firstNames }} {{ request.foundProxy.lastName }}
             </a>
         {% elseif request.processed %}
-            <a href="{{ path('app_procuration_manager_request_transform', {'id': request.id, 'action': 'detraiter', 'token': csrfToken}) }}"
+            <a href="{{ path('app_procuration_manager_request_transform', {'id': request.id, 'action': 'detraiter', 'token': csrf_token('request_action')}) }}"
```
Comment (thomasbisignani) [inline templates/procuration_manager/request.html.twig]: Simples quotes in array keys are useless

File: src/Controller/EnMarche/ProcurationController.php [hunk 5]
```diff
@@ -51,14 +53,11 @@ public function indexAction(Request $request): Response
      */
     public function voteAction(Request $request): Response
     {
-        $command = $this->getProcurationFlow()->getCurrentModel();
-
-        $form = $this->createForm(ProcurationVoteType::class, $command);
-        $form->handleRequest($request);
+        $form = $this->createForm(ProcurationVoteType::class, $this->get(ProcurationRequestSession::class)->getCurrentModel())
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: Action injection is possible here

File: templates/procuration_manager/request.html.twig [hunk 6]
```diff
@@ -17,12 +17,12 @@
...
                class="btn btn--small procuration-manager__request__process">
                 Marquer "En attente"
             </a>
         {% else %}
-            <a href="{{ path('app_procuration_manager_request_transform', {'id': request.id, 'action': 'traiter', 'token': csrfToken}) }}"
+            <a href="{{ path('app_procuration_manager_request_transform', {'id': request.id, 'action': 'traiter', 'token': csrf_token('request_action')}) }}"
```
Comment (thomasbisignani) [inline templates/procuration_manager/request.html.twig]: Same here, for simples quotes in array keys

File: src/Controller/EnMarche/ProcurationController.php [hunk 7]
```diff
@@ -73,18 +72,17 @@ public function voteAction(Request $request): Response
      */
     public function profileAction(Request $request): Response
     {
-        $command = $this->getProcurationFlow()->getCurrentModel();
+        $command = $this->get(ProcurationRequestSession::class)->getCurrentModel();
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: Action injection is possible here

File: src/Controller/EnMarche/ProcurationController.php [hunk 8]
```diff
@@ -99,18 +97,21 @@ public function profileAction(Request $request): Response
...
+        ;
 
         if ($form->isSubmitted() && $form->isValid()) {
             $manager = $this->getDoctrine()->getManager();
+
             $manager->persist($command);
             $manager->flush();
-
-            $this->getProcurationFlow()->reset();
+            // clean
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: I don't see the point of this comment. Why the session is clean-up would be a useful comment though.

File: src/Controller/EnMarche/ProcurationController.php [hunk 9]
```diff
@@ -99,18 +97,21 @@ public function profileAction(Request $request): Response
      */
     public function electionsAction(Request $request): Response
     {
-        $command = $this->getProcurationFlow()->getCurrentModel();
+        $procurationSession = $this->get(ProcurationRequestSession::class);
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: Action injection possible

File: src/Controller/EnMarche/ProcurationController.php [hunk 10]
```diff
@@ -99,18 +97,21 @@ public function profileAction(Request $request): Response
...
+        $command = $procurationSession->getCurrentModel();
```
Comment (B-Galati) [inline src/Controller/EnMarche/ProcurationController.php]: let's rename this variable to `$procuration` no ?

**Comment by thomasbisignani** [comment]
Nice refacto :ok_hand:

File: templates/procuration_manager/request.html.twig (hunk 4 above)
Comment (HeahDude) [reply templates/procuration_manager/request.html.twig]: I know this is not my code and I didn't want to change it everywhere... fixed in https://github.com/EnMarche/en-marche.fr/pull/1924/commits/c4be2d0911d19fb5148198bcf70cb395bd8b318f.

File: app/config/services/community.xml (hunk 1 above)
Comment (HeahDude) [reply app/config/services/community.xml]: Didn't no about that 👌, done. Thanks!

File: src/Controller/EnMarche/ProcurationController.php (hunk 2 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: Did not know it was set up, done. Thanks again!

File: src/Controller/EnMarche/ProcurationController.php (hunk 3 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: My bad, is this better now?

File: src/Controller/EnMarche/ProcurationController.php (hunk 5 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: Done.

File: src/Controller/EnMarche/ProcurationController.php (hunk 7 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: Done.

File: src/Controller/EnMarche/ProcurationController.php (hunk 9 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: Done.

File: src/Controller/EnMarche/ProcurationController.php (hunk 10 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: I've renamed it `$procurationRequest` thanks.

File: src/Controller/EnMarche/ProcurationController.php (hunk 8 above)
Comment (HeahDude) [reply src/Controller/EnMarche/ProcurationController.php]: Changed. Is it better?

**Pull request merged**